syntax = "proto3";

package BTL;

message MessageType {
	enum Message {
		HOSTINFO = 0;
		CLIENTINFO = 1;
		FILEINFO = 2;
		FILEDATA = 3;
		FILECACHE = 4;
	}
	Message message = 1;
	uint32 localPort = 2;
}

message HostInfo {
	string host = 1;
	uint32 port = 2;
	bool isSeeder = 3;
}

message ClientInfo {
	string remoteHost = 1;
	uint32 localPort = 2;
	bool isSeeder = 3;
	repeated HostInfo peers = 4;
}

message FileInfo {
	string filehash = 1;
	string fileName = 2;
	uint32 fileSize = 3;
}

message FileData {
	string fileName = 1;
	uint32 offset = 2;
	bytes data = 3;
}

message FileCache {
	bool is_seeder = 1;
	bool is_ack = 2;
	repeated uint32 cache = 3;
}
